internal/resolution/datasource/maven_registry.go

```diff
@@ -30,8 +31,12 @@ type MavenRegistryAPIClient struct {
 	// Cache fields
 	mu             *sync.Mutex
 	cacheTimestamp *time.Time // If set, this means we loaded from a cache
-	projects       *RequestCache[string, maven.Project]
-	metadata       *RequestCache[string, maven.Metadata]
+	responses      *RequestCache[string, response]
+}
+
+type response struct {
```

Comment:
I think of naming this as `mavenResponse` but this structure is also quite general